src/rosenfeld_fmt.cc

```diff
@@ -434,35 +431,69 @@ void RosenfeldFMT::computeSphericalDerivative(std::shared_ptr<State> state)
                 dF_dn3(idx) = 0.;
                 dF_dnv1(idx) = 0.;
                 dF_dnv2(idx) = 0.;
-
+                
+                double dF_dn3_1 = 0;
+                double dF_dn3_2 = 0;
                 // take integrals using quadrature
                 const auto r = mesh->center(idx);
-                const auto lower_bin = mesh->bin(0.);
-                const auto split_bin = mesh->bin(R-r);
-                const auto upper_bin = mesh->bin(r+R);
+                const auto lower = 0.;
+                const auto split = R-r;
+                const auto upper = r+R;
```

Comment:
I would be explicit about the type here:
```suggestion
                const double lower = 0.;
                const double split = R-r;
                const double upper = r+R;
```